### PR TITLE
FEATURE: Make pending unapproved posts visible to their authors

### PR DESCRIPTION
**Pull request by NickIvanter**
As discussed here: https://meta.discourse.org/t/how-to-make-the-queued-posts-visible-to-their-authors/39153

### CODE REVIEW

**Comment by coding-horror** [comment]
Thanks I do think this is better if the post is clearly marked as pending in some way. It is a lot of code though can you review @eviltrout after header is in?

**Comment by eviltrout** [comment]
Wow 47 files changed? This is going to take some time to review!

**Comment by NickIvanter** [comment]
It was a significant effort on our part. Thanks to my colleague @nnrd . We deployed it in production over a month ago and have been using since.

File: script/import_scripts/base.rb [hunk 1]
```diff
@@ -528,6 +528,52 @@ def create_bookmarks(results, opts={})
     [created, skipped]
   end
 
+  # Iterate through a list of "like" records to be imported.
+  # Takes a collection, and yields to the block for each element.
+  # Block should return a hash with the attributes for the "like".
+  # Required fields are :user_id and :post_id, where both ids are
+  # the values in the original datasource.
+  def create_likes(results, opts={})
```
Comment (xfalcox) [inline script/import_scripts/base.rb]: Is this related to the feature?

Comment (NickIvanter) [inline script/import_scripts/base.rb]: No, this must have gotten onto this branch by mistake. Good catch.

**Comment by eviltrout** [comment]
I responded on discourse meta because I think it's a better place for discussing this for now:

https://meta.discourse.org/t/how-to-make-the-queued-posts-visible-to-their-authors/39153/16?u=eviltrout

**Comment by coding-horror** [comment]
I like this change in theory but I think we may need to approach it a different way down the line?
